crates/sui-json-rpc/src/axum_router.rs

```diff
@@ -313,15 +311,20 @@ pub fn monitored_reroute(
         "sui_executeTransactionBlock" => {
             // add client IP arg to the params, as this is a router redirect
             // from `execute_transaction_block`, which does require the client IP
-            let parsed_value: Value = serde_json::from_str(
-                raw_params
-                    .unwrap_or_else(|| panic!("Expected params for executeTransactionBlock"))
-                    .get(),
-            )
-            .expect("Failed to parse jsonrpsee params");
+            let params = if let Some(params) = raw_params {
+                params.get()
+            } else {
+                return Err(SuiError::Unknown(String::from(
+                    "Params not found for executeTransactionBlock",
+                )));
+            };
+            let parsed_value: Value =
+                serde_json::from_str(params).expect("Failed to parse jsonrpsee params");
 
             let Value::Array(mut params_vec) = parsed_value else {
-                panic!("Expected a JSON array");
+                return Err(SuiError::Unknown(String::from(
+                    "Failed to parse jsonrpsee params: expected array",
```

Comment:
I believe @amnn was saying that this method used to handle non-array params?

Reply:
Yeah I have some more code locally to handle this case. But so that Im not playing wack a mole, wanna make sure the failsafe works. Will put that up once I've confirmed it works (doing that now)

Reply:
Should be good now